src/lazy/text/buffer.rs

```diff
@@ -715,7 +867,13 @@ impl<'data> TextBufferView<'data> {
     /// Matches a quoted symbol (`'foo'`).
     fn match_quoted_symbol(self) -> IonParseResult<'data, MatchedSymbol> {
         delimited(char('\''), Self::match_quoted_symbol_body, char('\''))
-            .map(|(_matched, contains_escaped_chars)| MatchedSymbol::Quoted(contains_escaped_chars))
```

Comment:
🗺️ As above, I've broken out the nested `(bool)` on the `MatchedSymbol` variant.